File: app/src/test/java/com/baro/helpers/JSONHelperTest.java
package com.baro.helpers;

import com.baro.constants.JSONEnum;

import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;

import java.util.HashMap;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

public class JSONHelperTest {


    @Test
    public void createJSONFromHashMapShouldReturnJSONObject() throws JSONException {
        HashMap<String, String> sampleHashMap = new HashMap<>();
        sampleHashMap.put(JSONEnum.COURSE_NAME_KEY.key, "Course Name");
        sampleHashMap.put(JSONEnum.COURSE_UUID_KEY.key, "Course UUID");

        JSONObject jsonTest = JSONHelper.createJSONFromHashMap(sampleHashMap);

        assertEquals(jsonTest.length(), 2);
        assertEquals(jsonTest.get(JSONEnum.COURSE_NAME_KEY.key), "Course Name");
        assertEquals(jsonTest.get(JSONEnum.COURSE_UUID_KEY.key), "Course UUID");

    }


    @Test
    public void createJSONFromStringShouldReturnJSONObjectWhenValidString() throws JSONException {
        String validJsonString = "{\"key1\": \"value\", \"key2\": 1}";

        JSONObject jsonTest = JSONHelper.createJSONFromString(validJsonString);

        assertEquals(jsonTest.length(), 2);
        assertEquals(jsonTest.get("key1"), "value");
        assertEquals(jsonTest.get("key2"), 1);
    }


    @Test
    public void createJSONFromStringShouldReturnNullWhenInvalidString()  {
        String invalidJsonString = "\"key1\": ";

        JSONObject jsonTest = JSONHelper.createJSONFromString(invalidJsonString);;
        assertNull(jsonTest);
    }
}

File: app/src/test/java/com/baro/helpers/JSONHelperTest.kt
package com.baro.helpers

import com.baro.constants.JSONEnum
import org.json.JSONException
import org.junit.Assert
import org.junit.Test
import java.util.*

class JSONHelperTest {
    @Test
    @Throws(JSONException::class)
    fun createJSONFromHashMapShouldReturnJSONObject() {
        val sampleHashMap = HashMap<String?, String?>()
        sampleHashMap[JSONEnum.COURSE_NAME_KEY.key] = "Course Name"
        sampleHashMap[JSONEnum.COURSE_UUID_KEY.key] = "Course UUID"
        val jsonTest = JSONHelper.createJSONFromHashMap(sampleHashMap)
        Assert.assertEquals(jsonTest!!.length().toLong(), 2)
        Assert.assertEquals(jsonTest!![JSONEnum.COURSE_NAME_KEY.key], "Course Name")
        Assert.assertEquals(jsonTest!![JSONEnum.COURSE_UUID_KEY.key], "Course UUID")
    }

    @Test
    @Throws(JSONException::class)
    fun createJSONFromStringShouldReturnJSONObjectWhenValidString() {
        val validJsonString = "{\"key1\": \"value\", \"key2\": 1}"
        val jsonTest = JSONHelper.createJSONFromString(validJsonString)
        Assert.assertEquals(jsonTest!!.length().toLong(), 2)
        Assert.assertEquals(jsonTest!!["key1"], "value")
        Assert.assertEquals(jsonTest!!["key2"], 1)
    }

    @Test
    fun createJSONFromStringShouldReturnNullWhenInvalidString() {
        val invalidJsonString = "\"key1\": "
        val jsonTest = JSONHelper.createJSONFromString(invalidJsonString)
        Assert.assertNull(jsonTest)
    }
}